all-actors/src/main/java/org/sunbird/NotificationRequestMapper.java

```diff
@@ -0,0 +1,41 @@
+package org.sunbird;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.pojo.NotificationRequest;
+
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class NotificationRequestMapper {
+
+    private static ObjectMapper mapper = new ObjectMapper();
+
+    private static Logger logger = LogManager.getLogger(NotificationRequestMapper.class);
+
+    public static List<NotificationRequest> toList(List<Map<String, Object>> request) throws BaseException {
+        if (request.isEmpty()) {
+            throw new BaseException("MANDATORY_PARAMETER_MISSING",
+                    MessageFormat.format(IResponseMessage.MANDATORY_PARAMETER_MISSING, JsonKey.NOTIFICATIONS),
+                    ResponseCode.CLIENT_ERROR.getCode());
+        }
+        List<NotificationRequest> sendNotificationList = new ArrayList<>();
+        for (Map<String, Object> map : request) {
+            sendNotificationList.add((getNotificationRequest(map)));
+        }
+        return sendNotificationList;
+    }
+
+
+    private static NotificationRequest getNotificationRequest(Map<String, Object> data) {
+        NotificationRequest notificationRequest = mapper.convertValue(data, NotificationRequest.class);
```

Comment:
What will happen if data is not able to map inside NotificationRequest  due to some unknow attribute.

Reply:
it will throw an IllegalArgumentException, I am catching this exception in Notification actor and throwing base exception